### ISSUE TITLE
SC2016: Does not apply on `unset`

### ISSUE THREAD

**Issue by stdedos** [koalaman/shellcheck#1186]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2016
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] ~~It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue~~

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
export SPLIT=('a' 'b')
unset 'SPLIT[${#SPLIT[@]}-1]'
```

#### Here's what shellcheck currently says:
```
$ shellcheck myscript
 
Line 3:
        unset 'SPLIT[${#SPLIT[@]}-1]'
              ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.

$ 
```

#### Here's what I wanted or expected to see:

(Nothing). `unset` will "expand the arguments itself". Or something else happens and this https://stackoverflow.com/a/8247497 works

**Comment by koalaman** [comment]
Fixed by @ngzhian, thanks!
